PropertyChangedAnalyzers/Analyzers/PropertyDeclarationAnalyzer.cs

```diff
@@ -82,7 +82,14 @@ private static void Handle(SyntaxNodeAnalysisContext context)
                     }
 
                     using var assignmentWalker = AssignmentWalker.Borrow(setter);
-                    if (assignmentWalker.Assignments.TryFirst(x => IsProperty(x.Left) && !x.Parent.IsKind(SyntaxKind.ObjectInitializerExpression), out var recursiveAssignment))
+                    if (assignmentWalker.Assignments.TryFirst(
+                        x =>
+                            IsProperty(x.Left) &&
+                            !x.Parent.IsKind(SyntaxKind.ObjectInitializerExpression)
+                            && SymbolEqualityComparer.Default.Equals(
+                                context.SemanticModel.GetDeclaredSymbol(propertyDeclaration, context.CancellationToken),
```

Comment:
GetDeclaredSymbol blows up for partial classes. There is a helper that I forgot the name of that handles that, maybe GetSymbolSafe, also it works on declarations and identifiers the same

Reply:
Good. I'd like to write a test that makes this blow up. Since C# doesn't have partial properties yet, what would it look like?

Reply:
Maybe it is never an issue, hard to tell. When it can be an issue is if an analyzer starts walking in one `partial` part and ends up in the other part where the semantic model is not valid.

Reply:
We can merge as is also of course

Reply:
Oh, I see GetSymbolSafe:

> Same as SemanticModel.GetSymbolInfo().Symbol but works when node is not in the syntax tree. Gets the semantic model for the tree if the node is not in the tree corresponding to semanticModel.

I think this means that `GetDeclaredSymbol` is an assertion that `propertyDeclaration` (`context.Node`) is in the syntax tree. I sort of like the assertion, but what do you think?

Reply:
(in chat, the reason is that `x.Left` on the following line came from the walker and might be from a different syntax tree)